File: osmToolsBasic/src/main/java/com/asamm/osmTools/cmdCommands/CmdSort.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package com.asamm.osmTools.cmdCommands;

import com.asamm.osmTools.Main;
import com.asamm.osmTools.mapConfig.ItemMap;

import java.io.File;

/**
 *
 * @author volda
 */
public class CmdSort extends Cmd{
    
    public CmdSort(ItemMap map) {
        super(map, ExternalApp.OSMOSIS);
    }

    public void createCmdSort() {
        addReadPbf();
        //addReadXml();
        addSort();
        addWritePbf(true);
    }

    private void addReadPbf() {
        if (!new File(getMap().getPathContour()).exists()) {
            throw new IllegalArgumentException("Soubor vrstevnic: " +
                    getMap().getPathContour()+ " neexistuje");
        }
        
        addCommand("--read-pbf");
        addCommand(getMap().getPathContour());
    }
    private void addReadXml() {
        if (!new File(getMap().getPathContour()).exists()){
            throw new IllegalArgumentException("Soubor vrstevnic: " +
                    getMap().getPathContour()+ " neexistuje");
        }
        addCommand("--read-xml");
        addCommand(getMap().getPathContour());
    }
    
   
    protected void addWritePbf (boolean omitMetadata) {
        addCommand("--write-pbf");
        addCommand(getMap().getPathContour()+".sorted");
        if (omitMetadata){
            addCommand("omitmetadata=true");
        }
    }
    private void addWriteXml () {
        addCommand("--wx");
        addCommand(getMap().getPathContour()+".sorted");
        addCommand("compressionMethod=bzip2");
        
    }

    public void rename () {
        //ocekavam, ze probehlo sortovani a prejmenuji mapu zpet na puvodni jmeno bez postfixu sorted
        File fSorted = new File(getMap().getPathContour() + ".sorted");
        File fUnsorted = new File(getMap().getPathContour());
              
        if (!fSorted.exists()){
            throw new IllegalArgumentException("Sorted contourline " +
                    getMap().getPathContour() + ".sorted does not exist.");
        }
        if (fUnsorted.exists()){
            fUnsorted.delete();
        }
        
        Main.LOG.info("Rename contour line: " + fSorted.getName() +
                " to: " + fUnsorted.getName());
        fSorted.renameTo(new File(getMap().getPathContour()));
    }
}

File: osmToolsBasic/src/main/java/com/asamm/osmTools/cmdCommands/CmdSort.kt
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package com.asamm.osmTools.cmdCommands

import com.asamm.osmTools.Main
import com.asamm.osmTools.mapConfig.ItemMap
import java.io.File

/**
 * @author volda
 */
class CmdSort(val map: ItemMap) : Cmd(ExternalApp.OSMOSIS), CmdOsmosis {
    fun createCmdSort() {
        addReadPbf()
        //addReadXml();
        addSort()
        addWritePbf(true)
    }

    private fun addReadPbf() {
        require(map.pathContour.toFile().exists()) {
            "File with contour lines " + map.pathContour + " does not exist"
        }

        addCommand("--read-pbf")
        addCommand(map.pathContour.toString())
    }

    protected fun addWritePbf(omitMetadata: Boolean) {
        addCommand("--write-pbf")
        addCommand(map.pathContour.toString() + ".sorted")
        if (omitMetadata) {
            addCommand("omitmetadata=true")
        }
    }

    private fun addWriteXml() {
        addCommand("--wx")
        addCommand(map.pathContour.toString() + ".sorted")
        addCommand("compressionMethod=bzip2")
    }

    fun rename() {
        //ocekavam, ze probehlo sortovani a prejmenuji mapu zpet na puvodni jmeno bez postfixu sorted
        val fSorted = File(map.pathContour.toString() + ".sorted")
        val fUnsorted = map.pathContour.toFile()

        require(fSorted.exists()) {
            "Sorted contourline " +
                    map.pathContour + ".sorted does not exist."
        }
        if (fUnsorted.exists()) {
            fUnsorted.delete()
        }

        Main.LOG.info(
            "Rename contour line: " + fSorted.name +
                    " to: " + fUnsorted.name
        )
        fSorted.renameTo(map.pathContour.toFile())
    }
}

